File: apiwznm/WznmQCalList.cpp
/**
	* \file WznmQCalList.cpp
	* API code for table TblWznmQCalList (implementation)
	* \author Alexander Wirthmueller
	* \date created: 25 Aug 2020
	* \date modified: 25 Aug 2020
	*/

#include "WznmQCalList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WznmQCalList
 ******************************************************************************/

WznmQCalList::WznmQCalList(
			const uint jnum
			, const string sref
			, const string srefIxVBasetype
			, const string titIxVBasetype
			, const string stubRefWznmMVersion
			, const string srefRefIxVTbl
			, const string titRefIxVTbl
			, const string stubRefUref
			, const string srefsInvIxWznmWArgtype
			, const string titsInvIxWznmWArgtype
			, const string srefsRetIxWznmWArgtype
			, const string titsRetIxWznmWArgtype
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->srefIxVBasetype = srefIxVBasetype;
	this->titIxVBasetype = titIxVBasetype;
	this->stubRefWznmMVersion = stubRefWznmMVersion;
	this->srefRefIxVTbl = srefRefIxVTbl;
	this->titRefIxVTbl = titRefIxVTbl;
	this->stubRefUref = stubRefUref;
	this->srefsInvIxWznmWArgtype = srefsInvIxWznmWArgtype;
	this->titsInvIxWznmWArgtype = titsInvIxWznmWArgtype;
	this->srefsRetIxWznmWArgtype = srefsRetIxWznmWArgtype;
	this->titsRetIxWznmWArgtype = titsRetIxWznmWArgtype;
};

bool WznmQCalList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WznmQCalList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "srefIxVBasetype", "typ", srefIxVBasetype);
		extractStringUclc(docctx, basexpath, "titIxVBasetype", "typ2", titIxVBasetype);
		extractStringUclc(docctx, basexpath, "stubRefWznmMVersion", "ver", stubRefWznmMVersion);
		extractStringUclc(docctx, basexpath, "srefRefIxVTbl", "ret", srefRefIxVTbl);
		extractStringUclc(docctx, basexpath, "titRefIxVTbl", "ret2", titRefIxVTbl);
		extractStringUclc(docctx, basexpath, "stubRefUref", "reu", stubRefUref);
		extractStringUclc(docctx, basexpath, "srefsInvIxWznmWArgtype", "iat", srefsInvIxWznmWArgtype);
		extractStringUclc(docctx, basexpath, "titsInvIxWznmWArgtype", "iat2", titsInvIxWznmWArgtype);
		extractStringUclc(docctx, basexpath, "srefsRetIxWznmWArgtype", "rat", srefsRetIxWznmWArgtype);
		extractStringUclc(docctx, basexpath, "titsRetIxWznmWArgtype", "rat2", titsRetIxWznmWArgtype);
	};

	return basefound;
};

/******************************************************************************
 class ListWznmQCalList
 ******************************************************************************/

ListWznmQCalList::ListWznmQCalList() {
};

ListWznmQCalList::ListWznmQCalList(
			const ListWznmQCalList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQCalList(*(src.nodes[i]));
};

ListWznmQCalList& ListWznmQCalList::operator=(
			const ListWznmQCalList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WznmQCalList(*(src.nodes[i]));

	return *this;
};

ListWznmQCalList::~ListWznmQCalList() {
	clear();
};

void ListWznmQCalList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWznmQCalList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WznmQCalList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWznmQCalList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WznmQCalList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WznmQCalList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WznmQCalList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

File: japiwznm/WznmQCalList.java
/**
  * \file WznmQCalList.h
  * Java API code for record of table TblWznmQCalList
  * \author Alexander Wirthmueller
  * \date created: 27 Aug 2020
  * \date modified: 27 Aug 2020
  */

package apiwznm;

import org.w3c.dom.*;
import sbecore.*;

public class WznmQCalList {
	
	public WznmQCalList(
				int jnum
				, String sref
				, String srefIxVBasetype
				, String titIxVBasetype
				, String stubRefWznmMVersion
				, String srefRefIxVTbl
				, String titRefIxVTbl
				, String stubRefUref
				, String srefsInvIxWznmWArgtype
				, String titsInvIxWznmWArgtype
				, String srefsRetIxWznmWArgtype
				, String titsRetIxWznmWArgtype
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.srefIxVBasetype = srefIxVBasetype;
		this.titIxVBasetype = titIxVBasetype;
		this.stubRefWznmMVersion = stubRefWznmMVersion;
		this.srefRefIxVTbl = srefRefIxVTbl;
		this.titRefIxVTbl = titRefIxVTbl;
		this.stubRefUref = stubRefUref;
		this.srefsInvIxWznmWArgtype = srefsInvIxWznmWArgtype;
		this.titsInvIxWznmWArgtype = titsInvIxWznmWArgtype;
		this.srefsRetIxWznmWArgtype = srefsRetIxWznmWArgtype;
		this.titsRetIxWznmWArgtype = titsRetIxWznmWArgtype;
	};

	public int jnum;
	public String sref;
	public String srefIxVBasetype;
	public String titIxVBasetype;
	public String stubRefWznmMVersion;
	public String srefRefIxVTbl;
	public String titRefIxVTbl;
	public String stubRefUref;
	public String srefsInvIxWznmWArgtype;
	public String titsInvIxWznmWArgtype;
	public String srefsRetIxWznmWArgtype;
	public String titsRetIxWznmWArgtype;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WznmQCalList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			srefIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "srefIxVBasetype", "typ", null, 0);
			titIxVBasetype = Xmlio.extractStringUclc(doc, basexpath, "titIxVBasetype", "typ2", null, 0);
			stubRefWznmMVersion = Xmlio.extractStringUclc(doc, basexpath, "stubRefWznmMVersion", "ver", null, 0);
			srefRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "srefRefIxVTbl", "ret", null, 0);
			titRefIxVTbl = Xmlio.extractStringUclc(doc, basexpath, "titRefIxVTbl", "ret2", null, 0);
			stubRefUref = Xmlio.extractStringUclc(doc, basexpath, "stubRefUref", "reu", null, 0);
			srefsInvIxWznmWArgtype = Xmlio.extractStringUclc(doc, basexpath, "srefsInvIxWznmWArgtype", "iat", null, 0);
			titsInvIxWznmWArgtype = Xmlio.extractStringUclc(doc, basexpath, "titsInvIxWznmWArgtype", "iat2", null, 0);
			srefsRetIxWznmWArgtype = Xmlio.extractStringUclc(doc, basexpath, "srefsRetIxWznmWArgtype", "rat", null, 0);
			titsRetIxWznmWArgtype = Xmlio.extractStringUclc(doc, basexpath, "titsRetIxWznmWArgtype", "rat2", null, 0);
			
			return true;
		};

		return false;
	};
};

